File: backend/src/main/java/com/example/backend/dto/VehicleOutputDTO.java
package com.example.backend.dto;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

/**
 * Data Transfer Object for vehicle output.
 */
@Data
@NoArgsConstructor
public class VehicleOutputDTO {
    private Long id;
    private String name;
    private Date creationDate;
    private Date lastModifiedDate;
    private Long version;
    private Long vehicleId;
    private String type;
    private int seats;
    private boolean canCarryWheelchair;
    private LocationDTO startLocation;
    private LocationDTO endLocation;
}

File: frontend/src/app/dtos/VehicleOutput.dto.ts
import {LocationDto} from "./Location.dto";

export interface VehicleOutputDto {
    id: number;
    name: string;
    creationDate: Date;
    lastModifiedDate: Date;
    version: number;
    vehicleId: number;
    type: string;
    seats: number;
    canCarryWheelchair: boolean;
    startLocation: LocationDto;
    endLocation: LocationDto;
}